alf/networks/network.py

```diff
@@ -188,8 +188,11 @@ def output_spec(self):
         can be obtained easily in other ways.
         """
         if self._output_spec is None:
+            training = self.training
+            self.eval()
```

Comment:
Similar switch between ```train()``` and ```eval()``` mode might be needed once BN is introduced (e.g. in rollout, play..) ?

Reply:
Done. Though I am not very satisfied how eval and train are implemented. They are using recursive calls to go through all submodules. It's kind of expensive. But it seems there no good way given how pytorch works.

Reply:
Right. This will add some extra cost. We can try to think of ways to improve it in the future.